diff --git a/libdnf/conf/ConfigMain.hpp b/libdnf/conf/ConfigMain.hpp
index 7dfdba3ef..704494593 100644
--- a/libdnf/conf/ConfigMain.hpp
+++ b/libdnf/conf/ConfigMain.hpp
@@ -23,6 +23,8 @@
 
 #ifdef LIBDNF_UNSTABLE_API
 
+#include <cstdint>
+
 #include "Config.hpp"
 #include "OptionBool.hpp"
 #include "OptionEnum.hpp"
diff --git a/libdnf/conf/OptionNumber.hpp b/libdnf/conf/OptionNumber.hpp
index f7a7b3d6e..717a3c926 100644
--- a/libdnf/conf/OptionNumber.hpp
+++ b/libdnf/conf/OptionNumber.hpp
@@ -23,6 +23,8 @@
 
 #ifdef LIBDNF_UNSTABLE_API
 
+#include <cstdint>
+
 #include "Option.hpp"
 
 #include <functional>
diff --git a/libdnf/dnf-advisoryref.cpp b/libdnf/dnf-advisoryref.cpp
index 6fe413e9f..4aae10c72 100644
--- a/libdnf/dnf-advisoryref.cpp
+++ b/libdnf/dnf-advisoryref.cpp
@@ -31,6 +31,7 @@
  * See also: #DnfContext
  */
 
+#include <cstdint>
 
 #include <solv/repo.h>
 #include <solv/util.h>
diff --git a/libdnf/dnf-conf.cpp b/libdnf/dnf-conf.cpp
index 800fbcd54..f4a98ee6c 100644
--- a/libdnf/dnf-conf.cpp
+++ b/libdnf/dnf-conf.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include "dnf-conf.h"
 
 #include "dnf-context.hpp"
diff --git a/libdnf/dnf-db.cpp b/libdnf/dnf-db.cpp
index 894703d34..e7ebe49b9 100644
--- a/libdnf/dnf-db.cpp
+++ b/libdnf/dnf-db.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include <glib.h>
 
 #include "dnf-db.h"
diff --git a/libdnf/dnf-reldep-list.cpp b/libdnf/dnf-reldep-list.cpp
index 8151a16c2..1663bad8a 100644
--- a/libdnf/dnf-reldep-list.cpp
+++ b/libdnf/dnf-reldep-list.cpp
@@ -16,6 +16,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include "dnf-reldep-list.h"
 #include "dnf-sack-private.hpp"
 #include "repo/solvable/Dependency.hpp"
diff --git a/libdnf/dnf-reldep.cpp b/libdnf/dnf-reldep.cpp
index 3991e2f17..0c3b7e4e1 100644
--- a/libdnf/dnf-reldep.cpp
+++ b/libdnf/dnf-reldep.cpp
@@ -16,6 +16,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include <solv/pool.h>
 
 #include "dnf-types.h"
diff --git a/libdnf/dnf-repo.cpp b/libdnf/dnf-repo.cpp
index 05570d9ae..7fbf3e728 100644
--- a/libdnf/dnf-repo.cpp
+++ b/libdnf/dnf-repo.cpp
@@ -33,6 +33,8 @@
  * See also: #DnfRepo
  */
 
+#include <cstdint>
+
 #include "conf/OptionBool.hpp"
 
 #include "dnf-context.hpp"
diff --git a/libdnf/dnf-sack-private.hpp b/libdnf/dnf-sack-private.hpp
index 898735348..53475a0dd 100644
--- a/libdnf/dnf-sack-private.hpp
+++ b/libdnf/dnf-sack-private.hpp
@@ -21,6 +21,8 @@
 #ifndef HY_SACK_INTERNAL_H
 #define HY_SACK_INTERNAL_H
 
+#include <cstdint>
+
 #include <stdio.h>
 #include <solv/pool.h>
 #include <vector>
diff --git a/libdnf/dnf-sack.cpp b/libdnf/dnf-sack.cpp
index 0e2613241..fdcd727cc 100644
--- a/libdnf/dnf-sack.cpp
+++ b/libdnf/dnf-sack.cpp
@@ -31,6 +31,7 @@
  * See also: #DnfContext
  */
 
+#include <cstdint>
 
 #include <algorithm>
 #include <assert.h>
diff --git a/libdnf/hy-goal.cpp b/libdnf/hy-goal.cpp
index 8dc60437b..749b4773a 100644
--- a/libdnf/hy-goal.cpp
+++ b/libdnf/hy-goal.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include <assert.h>
 #include <errno.h>
 #include <stdio.h>
@@ -58,6 +60,8 @@ extern "C" {
 
 #include "utils/bgettext/bgettext-lib.h"
 
+#include <cstdint>
+
 #define BLOCK_SIZE 15
 
 
diff --git a/libdnf/hy-iutil.cpp b/libdnf/hy-iutil.cpp
index 497c560d4..82693f3b4 100644
--- a/libdnf/hy-iutil.cpp
+++ b/libdnf/hy-iutil.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include <assert.h>
 #include <errno.h>
 #include <dirent.h>
diff --git a/libdnf/hy-package.cpp b/libdnf/hy-package.cpp
index ba3f32cc5..9b9b5d135 100644
--- a/libdnf/hy-package.cpp
+++ b/libdnf/hy-package.cpp
@@ -29,6 +29,8 @@
  * See also: #DnfContext
  */
 
+#include <cstdint>
+
 #include "libdnf/utils/utils.hpp"
 
 #include <algorithm>
diff --git a/libdnf/hy-query.cpp b/libdnf/hy-query.cpp
index 8bb981176..bdec6b869 100644
--- a/libdnf/hy-query.cpp
+++ b/libdnf/hy-query.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include <assert.h>
 #include "hy-goal-private.hpp"
 #include "hy-iutil-private.hpp"
diff --git a/libdnf/hy-selector.cpp b/libdnf/hy-selector.cpp
index 6ecb34d9c..a3709af1b 100644
--- a/libdnf/hy-selector.cpp
+++ b/libdnf/hy-selector.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include "sack/selector.hpp"
 
 HySelector
diff --git a/libdnf/transaction/Item.hpp b/libdnf/transaction/Item.hpp
index fd93106c7..25402c752 100644
--- a/libdnf/transaction/Item.hpp
+++ b/libdnf/transaction/Item.hpp
@@ -21,6 +21,8 @@
 #ifndef LIBDNF_TRANSACTION_ITEM_HPP
 #define LIBDNF_TRANSACTION_ITEM_HPP
 
+#include <cstdint>
+
 #include <memory>
 #include <string>
 
diff --git a/libdnf/transaction/Swdb.hpp b/libdnf/transaction/Swdb.hpp
index 5b2342c88..7b13167ae 100644
--- a/libdnf/transaction/Swdb.hpp
+++ b/libdnf/transaction/Swdb.hpp
@@ -21,6 +21,8 @@
 #ifndef LIBDNF_TRANSACTION_SWDB_HPP
 #define LIBDNF_TRANSACTION_SWDB_HPP
 
+#include <cstdint>
+
 #include <map>
 #include <memory>
 #include <solv/pooltypes.h>
diff --git a/libdnf/transaction/Transaction.hpp b/libdnf/transaction/Transaction.hpp
index a4aba8f3c..b9d909489 100644
--- a/libdnf/transaction/Transaction.hpp
+++ b/libdnf/transaction/Transaction.hpp
@@ -21,6 +21,8 @@
 #ifndef LIBDNF_TRANSACTION_TRANSACTION_HPP
 #define LIBDNF_TRANSACTION_TRANSACTION_HPP
 
+#include <cstdint>
+
 #include <memory>
 #include <set>
 #include <string>
diff --git a/libdnf/transaction/private/Repo.hpp b/libdnf/transaction/private/Repo.hpp
index 573603746..2c18a298e 100644
--- a/libdnf/transaction/private/Repo.hpp
+++ b/libdnf/transaction/private/Repo.hpp
@@ -21,6 +21,8 @@
 #ifndef LIBDNF_TRANSACTION_REPO_HPP
 #define LIBDNF_TRANSACTION_REPO_HPP
 
+#include <cstdint>
+
 #include <map>
 #include <memory>
 #include <string>
diff --git a/libdnf/utils/sqlite3/Sqlite3.cpp b/libdnf/utils/sqlite3/Sqlite3.cpp
index c7d03c357..6eb29eb80 100644
--- a/libdnf/utils/sqlite3/Sqlite3.cpp
+++ b/libdnf/utils/sqlite3/Sqlite3.cpp
@@ -19,6 +19,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <cstdint>
+
 #include "Sqlite3.hpp"
 
 void
diff --git a/libdnf/utils/sqlite3/Sqlite3.hpp b/libdnf/utils/sqlite3/Sqlite3.hpp
index 3a7da23cc..c9cb4715f 100644
--- a/libdnf/utils/sqlite3/Sqlite3.hpp
+++ b/libdnf/utils/sqlite3/Sqlite3.hpp
@@ -25,6 +25,8 @@
 #include "../../error.hpp"
 #include "../../log.hpp"
 
+#include <cstdint>
+
 #include <sqlite3.h>
 
 #include <map>
